File: extractor-appender/appender-sql/src/main/java/ru/mydesignstudio/database/metadata/extractor/database/view/referenced/ViewReferencedExtractor.java
package ru.mydesignstudio.database.metadata.extractor.database.view.referenced;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import ru.mydesignstudio.database.metadata.extractor.database.ExtractHelper;
import ru.mydesignstudio.database.metadata.extractor.database.TableMetadataAppender;
import ru.mydesignstudio.database.metadata.extractor.extract.result.TableMetadata;
import ru.mydesignstudio.database.metadata.extractor.extract.result.ViewReferencedModel;
import ru.mydesignstudio.database.metadata.extractor.resource.StringResource;

@Slf4j
@Component
public class ViewReferencedExtractor implements TableMetadataAppender {
  @StringResource("classpath:sql/extract_views_referenced.sql")
  private String query;

  @Autowired
  private ExtractHelper helper;

  @Override
  public void append(TableMetadata metadata, String schemaName, String tableName) {
    log.info("Extracting referenced views for table {} in schema {}", tableName, schemaName);

    metadata.setViewsReferenced(helper.extract(query, new Object[]{ tableName, schemaName }, ViewReferencedModel.class));

    log.debug("Extracted {} referenced views", metadata.getViewsReferenced().size());
  }
}

File: extractor-source/source-oracle/src/main/kotlin/ru/mydesignstudio/database/metadata/extractor/table/extractor/ViewReferencedExtractor.kt
package ru.mydesignstudio.database.metadata.extractor.table.extractor

import org.springframework.stereotype.Component
import ru.mydesignstudio.database.metadata.extractor.ExtractorBase
import ru.mydesignstudio.database.metadata.extractor.connection.DataSourceProvider
import ru.mydesignstudio.database.metadata.extractor.extract.helper.QueryProvider
import ru.mydesignstudio.database.metadata.extractor.extract.parameters.source.MetadataObject
import ru.mydesignstudio.database.metadata.extractor.extract.parameters.source.MetadataSchema
import ru.mydesignstudio.database.metadata.extractor.extract.parameters.source.MetadataSourceConnection
import ru.mydesignstudio.database.metadata.extractor.extract.result.ViewReferencedModel

@Component
class ViewReferencedExtractor (
        private val dataSourceProvider: DataSourceProvider,
        private val queryProvider: QueryProvider) : ExtractorBase() {
    fun extract(metadata: MetadataObject, schema: MetadataSchema, connection: MetadataSourceConnection): Collection<ViewReferencedModel> {
        val query = queryProvider.provide("classpath:sql/extract_views_referenced.sql")
        val dataSource = dataSourceProvider.provide(connection)
        return extract(dataSource, query, listOf(metadata.name, schema.name)) { rs ->
            ViewReferencedModel(
                    referencingObject = getString(rs, "referencing_object"),
                    referencingType = getString(rs, "referencing_type"),
                    referencedObject = getString(rs, "referenced_object"),
                    referencedType = getString(rs, "referenced_type")
            )
        }
    }
}